[WIP] Add Additional Resources page

tekd opened the pull request:
Here is an additional resources page in progress. I still have to clean up some of the files (particularly the yaml file), but any feedback/instruction is appreciated.

tekd committed:
Add AIII open source resource yaml file and add to build.py

tekd committed:
Change filepath for images

tekd committed:
Add projects-list styles for APPIII

tekd committed:
Add js for project-list/tiles/modals

tekd committed:
Add resource images

tekd committed:
Center text in resource dividers

tekd committed:
Change columns for appendix content

tekd committed:
Hyperlink resources instead of listing URL separately

tekd committed:
Add table to APP II

tekd committed:
Hyperlink APP III content and arrange in columns

tekd committed:
Clean up project list scss

tekd committed:
Clean up modals js and html

tekd committed:
Clean up open data resources yaml

toolness commented on data/open-data-resources.yaml:
```diff
@@ -0,0 +1,119 @@
+- title: '1. ePSIplatform’s Cases'
```
Is it possible and/or useful to automatically generate the number from the item's index in the list of open data resources? Dunno, might be over-engineering, but figured I'd mention it.

toolness commented on data/open-data-resources.yaml:
```diff
@@ -0,0 +1,119 @@
+  countries_covered: 'Burkina Faso, Mexico, Morocco, Kyrgyzstan, Moldova, Macedonia and South Africa'
+  contact: '<a href="https://theodi.org/team/tom-heath"><span class="CharOverride-2">Tom Heath</span></a>, Head of Research, ODI; (@tommyh)</p>
+                  <p class="text">Project URL: <a href="https://theodi.org/odp4d"><span class="CharOverride-2">https://theodi.org/odp4d</span></a'
+  case_studies_url: '<a href="https://theodi.org/case-studies"><span class="CharOverride-2">https://theodi.org/case-studies</span></a>'
```
Ummm this does not appear to be a URL... ?

In general, for fields that can include HTML, I recommend adding `_html` to the end of the field name (e.g. `contact_html`) so that it's as clear as possible to anyone reading it, but it's up to you.

tekd committed:
Clean up data and add '_html' when necessary

tekd committed:
Revise App III to conform with yaml edits and remove close material icon

tekd committed:
Remove hardcoded numbers in APP III resource titles

tekd commented:
I pushed up some new changes:
- yaml clean up. This should look much cleaner.
- Removed the hardcoded numbers in titles and appending index+1 through a loop
- added _html when needed: The html in the file isn't ideal, but I am not sure how else to handle the footnotes/reference links and contact links. Should I style them as footnotes by adding the "e-fn-ref" class even though they are just links?

I ended up using a table for the glossary and 2 columns for the links after all. The Travis tests will fail me again, but I confirmed with @mocxd that there were footnote link issues that were causing this.

tekd committed:
Add alt text for APP III image tiles

mocxd commented on data/open-data-resources.yaml:
```diff
@@ -0,0 +1,116 @@
+- title: "Follow the Money "
+  image:  "static/img/resources/Image10195.png"
+  who_html: 'A global network of more than 100 organizations supported by ONE and the Open Knowledge Foundation. Other partners include the Transparency &amp; Accountability Initiative, the World Bank, Development Initiatives, and the Revenue Watch Institute.'
+  what_html: 'The network enables organizations working on “various aspects of the Follow the Money agenda – from resource availability, to spending, to service delivery and results in particular sectors to share information, campaign for the inclusion of open data principles in transparency policies, and identify opportunities for collaboration to drive policy change, including through Follow th ...
```
sort of preference thing, but multiline strings with `>` might be more readable here for these long html fields in the yaml

ghost commented:
as far as i'm concerned, _small_ snippets of html embedded within yaml is a perfectly acceptable use case, and is much cleaner looking when multiline strings are used.

in order to insert large html sections into templates, like in the example of our case study pages, we have child templates which consist nearly entirely of `{% block %}`s to insert long html sections into the parent template, while the shorter bits of metadata about the case studies are still contained within the yaml.  this a slightly nicer solution for merging large html blocks into the structure set up by the yaml, but it is not without drawbacks (elaborated below), and it is way overkill for tiny snippets of html for these modals imo.

the primary drawback that makes this method not a great generic solution is that it is actually a parallel data structure to the structure of the site.  this was due to a hacky time-bereft approach, and could be greatly improved. you'll notice this sneaky line at the top of each child template: `{% set case = cases[13] %}` which sets the metadata information from the yaml for the rendered template and is essentially a hack which is highly dependent on the structure of the yaml itself and prone to breakage.

on the other end of the spectrum, we have a function within `build.py` to handle building multiple derivative templates based on a single parent template ( https://github.com/GovLab/open-data-impact-case-studies/blob/master/build.py#L47 ), which we use on other sites, like the www, in order to build a lot of these derived template pages with the proper metadata.

one approach to modernizing this process would be to expand on this function in order to be able to source child '`block`' templates, a parent template, and metadata and merge this information together into nicely packaged derivative templates, but i would rather it be done in the new js/gulp based process instead of with python. something i have already started in gulp in order to move the original `render_detail_pages` to the new stack and which could be the base code for this functionality is here: https://github.com/GovLab/furyosa/blob/csvdev/gulpfile.js#L158

and yes, we can probably safely merge while ignoring travis' complaints

tekd committed:
Add revised table image file for App. I and insert into html

tekd committed:
Remove unneccesary images

tekd committed:
Remove image height from appIII modal links

tekd committed:
Clean up html tags and fix columns and rows

tekd committed:
Add hr to separate app-4 names

tekd committed:
Replace World Bank Group Logo in App3

tekd committed:
Change top section color and add dark blue to sass variables for components

tekd commented:
FYI all changes on this resources branch have been merged with the key-findings branch in #4 so we can close this.
